src/osdc/Objecter.cc

```diff
@@ -3139,8 +3139,10 @@ void Objecter::_finish_op(Op *op, int r)
 
   // op->session->lock is locked unique or op->session is null
 
-  if (!op->ctx_budgeted && op->budgeted)
-    put_op_budget(op);
+  if (!op->ctx_budgeted && op->budget) {
```

Comment:
what if the op has no budget? in that case, we should still `op_throttle_ops.put(1);`, shouldn't we?